src/formatDate.js

```diff
@@ -50,7 +50,38 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fullDate = {};
+  const finallDate = [];
+  const dateArr = date.split(fromFormat[3]);
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    const from = fromFormat[i];
+    const initialDate = dateArr[i];
+
+    if (from === 'YY') {
+      if (initialDate < 30) {
+        fullDate['YYYY'] = `20${initialDate}`;
+      }
+
+      if (initialDate >= 30) {
+        fullDate['YYYY'] = `19${initialDate}`;
+      }
+    }
+
+    fullDate[from] = initialDate;
+  }
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    const to = toFormat[i];
+
+    if (to === 'YY') {
+      finallDate.push(fullDate['YYYY'].slice(2));
+    } else {
+      finallDate.push(fullDate[to]);
+    }
+  }
+
+  return finallDate.join(toFormat[3]);
```

Comment:
You can use separate variable to toFormat[3], because you use it twice,it will make your code more understandable

Reply:
agree